File: backend/entities/vehicle.go
package entities

import (
	"time"
)

type Vehicle struct {
	ID            int       `json:"id" gorm:"primaryKey;autoIncrement:true"`
	ModelName     string    `json:"model_name" gorm:"not null"`
	Matriculation string    `json:"matriculation" gorm:"not null"`
	Seat          int       `json:"seat" gorm:"not null"`
	Type          string    `json:"type" gorm:"not null"`
	IsVerified    bool      `json:"is_verified"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
	UserID        int       `json:"user_id"`
	User          User      `json:"user" gorm:"foreignkey:UserID"`
}

File: frontend/lib/models/vehicle.dart
class Vehicle {
  final int? id;
  String matriculation;
  String modelName;
  int seat;
  String type;
  bool? isVerified;
  final String? createdAt;
  final String? updatedAt;

  Vehicle({
    this.id,
    required this.matriculation,
    required this.modelName,
    required this.seat,
    required this.type,
    this.isVerified,
    this.createdAt,
    this.updatedAt,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'],
      matriculation: json['matriculation'],
      modelName: json['model_name'],
      seat: json['seat'],
      type: json['type'],
      isVerified: json['is_verified'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }

  toJson() {
    return {
      'id': id,
      'matriculation': matriculation,
      'model_name': modelName,
      'seat': seat,
      'type': type,
      'is_verified': isVerified,
      'created_at': createdAt,
      'updated_at': updatedAt,
    };
  }
}
